packages/netlify-cms-backend-github/src/API.ts

```diff
@@ -52,6 +52,7 @@ export interface Config {
   originRepo?: string;
   squashMerges: boolean;
   initialWorkflowStatus: string;
+  includeCookiesInRequests?: boolean;
```

Comment:
This code in this file is applied when using the GitHub backend directly (without git-gateway).
I don't think we would like to have the new logic when not using git-gateway

Reply:
This is only necessary for the github backend to accept the options for this. The `include_cookies_with_requests` option doesn't work unless you're using the git gateway.

The code has to be changed in these files to support this feature, because the requests are made in these files.

Reply:
Using the following configuration with the current code changes:

```yaml
backend:
  name: github
  repo: owner/repo
  include_cookies_in_requests: true
```

Will set `credentials: include` to requests sent directly to the GitHub API. I don't think it is desirable, same for GitLab.
Since both the GitHub API class and the GitLab API class are extended here https://github.com/netlify/netlify-cms/blob/3029d5b5fbebd0cd0742caf56c622dfbf0dfb1df/packages/netlify-cms-backend-git-gateway/src/GitHubAPI.ts#L11 and here https://github.com/netlify/netlify-cms/blob/3029d5b5fbebd0cd0742caf56c622dfbf0dfb1df/packages/netlify-cms-backend-git-gateway/src/GitLabAPI.ts#L7 accordingly, I think the logic to modify the `credentials` option should only be implemented in those classes.